static/src/javascripts-legacy/projects/common/modules/atoms/story-questions.js

```diff
@@ -19,111 +19,98 @@ define([
+        $('.js-storyquestion-email-question-button', component).addClass('is-hidden');
+        $('.js-storyquestion-email-nospam', component).addClass('is-hidden');
+
+        $('.js-storyquestion-email-enter', component).removeClass('is-hidden');
+        $('.js-storyquestion-email-signup-form', component).removeClass('is-hidden');
+    }
+
     function submitSignUpForm(event) {
         event.preventDefault();
 
         var answersEmailSignupForm = event.currentTarget;
         var email = answersEmailSignupForm.elements['email'];
         var listId = answersEmailSignupForm.listId;
-        var questionId = answersEmailSignupForm.dataset.questionId;
-
-        if (email && listId && questionId) {
 
+        if (email && listId) {
             fetch(config.page.ajaxUrl + '/story-questions/answers/signup', {
                 mode: 'cors',
                 method: 'POST',
                 body: {email : email.value, listId: listId.value}
             })
             .then(function (response) {
                 if (response.ok) {
-                    var submissionContainerEl = answersEmailSignupForm.parentElement;
-                    var thankyouMessage = document.getElementById('js-final-thankyou-message-' + questionId);
-
-                    if (submissionContainerEl && thankyouMessage) {
-                        submissionContainerEl.classList.add('is-hidden');
-                        thankyouMessage.classList.remove('is-hidden');
-                    }
+                    var component = event.currentTarget.closest('.js-view-tracking-component');
+                    $('.js-storyquestion-email-done', component).removeClass('is-hidden');
+                    $('.js-storyquestion-email-question', component).addClass('is-hidden');
+                    $('.js-storyquestion-email-signup-form', component).addClass('is-hidden');
+                    $('.js-storyquestion-email-enter', component).addClass('is-hidden');
+                    $('.js-storyquestion-email-nospam', component).addClass('is-hidden');
                 }
             });
         }
     }
 
     return {
         init: function() {
-            var askQuestionLinks = $('.js-ask-question-link');
-            var isEmailSubmissionReadyElement = document.getElementById('js-storyquestion-is-email-submission-ready');
-
-            var isEmailSubmissionReady = false;
-
-            if (isEmailSubmissionReadyElement) {
-                isEmailSubmissionReady = isEmailSubmissionReadyElement.dataset.isEmailSubmissionReady ? isEmailSubmissionReadyElement.dataset.isEmailSubmissionReady : false;
-            }
 
-            askQuestionLinks.each(function (el) {
-                bean.on(el, 'click', function(event) {
-                    askQuestion(event, isEmailSubmissionReady)
-                    this.classList.add('is-clicked')
-                });
+            $('.js-storyquestion-email-question-button').each(function(el) {
+                bean.on(el, 'click', emailMe);
```

Comment:
let's remove these listeners once the job is done, so that the browser can clean up memory.